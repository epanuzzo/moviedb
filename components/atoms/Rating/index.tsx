import React from "react";

type RatingProps = {
  isLarge?: boolean;
  text: number | string;
};

/**
 * Rating component displays a rating badge with customizable size.
 *
 * @component
 * @param {object} props - Component props
 * @param {boolean} [props.isLarge=false] - Determines the size of the rating badge. If true, a larger size is applied.
 * @param {string} props.text - The text to display inside the rating badge.
 * @returns {JSX.Element} The rendered rating badge component.
 */
const Rating: React.FC<RatingProps> = ({ isLarge = false, text }) => {
  const size = isLarge ? "w-12 h-12" : "w-10 h-10";
  return (
    <div
      className={`absolute top-4 left-4 bg-white text-black rounded-full flex items-center justify-center ${size}`}
    >
      <p className="text-sm font-bold">{text}</p>
    </div>
  );
};

export default Rating;
