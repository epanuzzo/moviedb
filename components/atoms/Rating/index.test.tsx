import React from "react";
import renderer from "react-test-renderer";
import Rating from "./index";

describe("Rating", () => {
  it("renders component", () => {
    const tree = renderer.create(<Rating isLarge text="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders component with alt props", () => {
    const tree = renderer.create(<Rating text="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
