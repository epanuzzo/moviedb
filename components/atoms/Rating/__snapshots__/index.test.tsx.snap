// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders component 1`] = `
<div
  className="absolute top-4 left-4 bg-white text-black rounded-full flex items-center justify-center w-12 h-12"
>
  <p
    className="text-sm font-bold"
  >
    test
  </p>
</div>
`;

exports[`Rating renders component with alt props 1`] = `
<div
  className="absolute top-4 left-4 bg-white text-black rounded-full flex items-center justify-center w-10 h-10"
>
  <p
    className="text-sm font-bold"
  >
    test
  </p>
</div>
`;
